### PR TITLE
feat(demo): improve styling and add italian

### PR DESCRIPTION
**Pull request by casungo**
This pull request includes several updates and improvements to the i18n demo:

- Revamped the header and language switch components
- Created new reusable components
- Updated page layouts and content
- Added Italian language support
- Removed old test pages
- Updated the README with the latest information

The French translations were machine-generated as a starting point, but should be reviewed by a native speaker before merging (you :P ). The italian translation is fine since I'm native in Italian.
Description, author and body fields for blog posts have not been implemented, maybe you can give it a try when you have time. I don't know enough of Astro to do that 😄

### CODE REVIEW

File: demo/astro.config.ts [hunk 1]
```diff
@@ -10,16 +10,19 @@ export default defineConfig({
...
 					fr: "/a-propos",
+					it: "/chi-siamo",
 				},
 				"/blog": {
 					fr: "/le-blog",
+					it: "/blog",
 				},
 				"/blog/[slug]": {
 					fr: "/le-blog/[slug]",
+					it: "/blog/[slug]",
```
Comment (florian-lefebvre) [inline demo/astro.config.ts]:
```suggestion
```
Since it's the same, it can be omitted!

File: demo/src/components/Footer.astro [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+import { t } from "i18n:astro";
+
+---
+<footer class="bg-gray-300 p-4 sticky bottom-0 flex justify-center items-center text-center" style="position:absolute;width:100%;padding-bottom:2rem;">
```
Comment (florian-lefebvre) [inline demo/src/components/Footer.astro]: Can we use only tailwind and not inline styles?

File: demo/src/components/Footer.astro [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+import { t } from "i18n:astro";
+
+---
```
Comment (florian-lefebvre) [inline demo/src/components/Footer.astro]:
```suggestion
import { t } from "i18n:astro";
---

```

File: demo/src/locales/en/common.json [hunk 4]
```diff
@@ -1,5 +1,10 @@
 {
 	"home": "Home",
+	"hero": "Welcome to I18n for Astro Demo, a website that showcases I18n for Astro",
```
Comment (florian-lefebvre) [inline demo/src/locales/en/common.json]:
```suggestion
	"hero": "Welcome to I18n for Astro Demo",
```

File: demo/src/locales/fr/common.json [hunk 5]
```diff
@@ -1,5 +1,10 @@
 {
 	"home": "Accueil",
+	"hero": "Bienvenue sur la démo d'I18n pour Astro, un site web qui présente I18n pour Astro",
```
Comment (florian-lefebvre) [inline demo/src/locales/fr/common.json]:
```suggestion
	"hero": "Bienvenue sur la démo d'I18n pour Astro",
```

File: demo/src/locales/it/common.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+  "home": "Home",
+  "hero": "Benvenuti alla demo di I18n per Astro, un sito web che mostra le funzionalità di I18n per Astro",
```
Comment (florian-lefebvre) [inline demo/src/locales/it/common.json]:
```suggestion
  "hero": "Benvenuti alla demo di I18n per Astro",
```

File: demo/src/pages/test.astro
Comment (florian-lefebvre) [inline demo/src/pages/test.astro:1]: The test pages are actually important! Basically they aim to show that pages outside of `src/routes` can still benefit from `i18n:astro`. There's probably a better way to do it tho!

File: demo/src/routes/about.astro [hunk 7]
```diff
@@ -1,10 +1,15 @@
...
 
 const title = t("about");
 ---
 
 <Layout {title}>
-  <h1>{title}</h1>
+  <div class="justify-center items-center">
+    <HeroMessage message="aboutMessage" />
+    <HeroMessage message="opensource" />
+    <a href="https://github.com/astrolicious/i18n/tree/main/demo" class="underline"><HeroMessage message="opensourceCTA" /></a>
```
Comment (florian-lefebvre) [inline demo/src/routes/about.astro:13]: Don't remember if it's there already but can you add a link in the header as well?

**Review by florian-lefebvre — CHANGES_REQUESTED** [review]
Awesome contribution thank you! Can you add `sharp` to the demo dependencies to make it work on netlify? Also left a few reviews

File: demo/src/pages/test.astro
Comment (casungo) [reply demo/src/pages/test.astro:1]: Oh I complitely missed this. Will fix!

File: demo/src/components/Footer.astro (hunk 2 above)
Comment (casungo) [reply demo/src/components/Footer.astro]: oops that was supposed to be temporary while i figured another way to do it :P
in my todo list :)

File: demo/src/routes/about.astro (hunk 7 above)
Comment (casungo) [reply demo/src/routes/about.astro:13]: Sure!

**Comment by casungo** [comment]
this should cover all of the problems :)

File: demo/src/components/Header.astro [hunk 8]
```diff
@@ -2,14 +2,17 @@
...
-        <a href={getLocalePath("/about")}>{t("about")}</a>
-        <a href={getLocalePath("/blog")}>{t("blog")}</a>
-      </div>
-      <LocaleSwitcher />
-    </header>
\ No newline at end of file
+<header class="bg-gray-300 p-2 sticky top-0 flex items-center justify-between flex-col sm:flex-row">
+  <a href="https://github.com/astrolicious/i18n/tree/main/demo">
+    <div class="flex items-center p-3">
+      <img src="../../src/content/astrolicious.png" alt="astrolicious" class="rounded-lg mr-3" >
```
Comment (florian-lefebvre) [inline demo/src/components/Header.astro]: You can't use like so. Move the image to the public folder and use an absolute path

**Review by florian-lefebvre — CHANGES_REQUESTED** [review]

**Comment by florian-lefebvre** [comment]
Just saw that the language switcher is broken for some reason * sigh * (nothing to do with the demo don't worry)

**Review by florian-lefebvre — APPROVED** [review]
Awesome thank you!

**Pull request merged**